modulo2/projeto-labefy/labefy/src/App.js

```diff
@@ -0,0 +1,271 @@
+font-family: Ariel;
+text-align: justified;
+align-items: justified;
+margin: 10px;
+padding: 0;
+`
+
+const ContainerComandos = styled.div`
+display: flex;
+flex-direction: column;
+height: 100%;
+
+input {
+margin-bottom: 20px;
+margin: 10px;
+}
+`
+
+const ButtonStyle = styled.button`
+padding: 2px;
+background-color: white;
+border-radius: 5px;
+color: black;
+font-size: 15px;
+font-weight: bold;
+border: 2px solid black;
+&:hover{
+  background: black;
+  color: white;
+  cursor: pointer;
+}
+`
+
+export default class App extends React.Component {
+  state = {
+    playlistName: "",
+    playlists: [],
+    playlistTracks: [],
+    music: "",
+    artista: "",
+    urlLink: "",
+    playlistId: ""
+  }
+
+  componentDidMount() {
+    this.getAllPlaylists()
+  }
+
+  getAllPlaylists = () => {
+    axios.get("https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists", {
+      headers: {
+        Authorization: "ederson-mello-carver"
+      }
+    })
+
+      .then((response) => {
+        this.setState({ playlists: response.data.result.list })
+        console.log(response)
+      })
+
+      .catch((err) => {
+        alert(err.response.data)
+      })
+  }
+
+  createPlaylist = () => {
+    const body = {
+      name: this.state.playlistName,
+    }
+    axios.post("https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists", body, {
+      headers: {
+        Authorization: "ederson-mello-carver"
+      }
+    })
+
+      .then(() => {
+        this.setState({ playlistName: "" })
+        this.getAllPlaylists()
+        alert("Playlist Criada!")
+      })
+      .catch(() => {
+        alert("Erro")
+      })
+  }
+
+  deletePlaylist = (playlistId) => {
+    axios.delete(`https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists/${playlistId}`, {
+      headers: {
+        Authorization: "ederson-mello-carver"
+      }
+    })
+      .then(() => {
+        alert("Playlist Deletada")
+        this.getAllPlaylists()
+      })
+      .catch((err) => {
+        alert(err.response.data)
+      })
+  }
+
+  getPlaylistsTracks = (id) => {
+    axios.get(`https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists/${id}/tracks`, {
+      headers: {
+        Authorization: 'ederson-mello-carver'
+      }
+    })
+      .then((response) => {
+        this.setState({ playlistTracks: response.data.result.tracks, playlistId: id })
+      })
+      .catch((err) => {
+        alert(err.response.data)
+      })
+  }
+
+  addTrackToPlaylist = () => {
+    const body = {
+      name: this.state.music,
+      artist: this.state.artista,
+      url: this.state.urlLink
+    }
+    axios.post(`https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists/${this.state.playlistId}/tracks`, body, {
+      headers: {
+        Authorization: "ederson-mello-carver"
+      }
+    })
+      .then((response) => {
+        this.setState({
+          music: "",
+          artista: "",
+          urlLink: "",
+        })
```

Comment:
Mandou muito bem usando o this.setState uma única vez para alterar todos os estados. show!